nixos/modules/testing/minimal-kernel.nix

```diff
@@ -1,28 +1,42 @@
 { config, pkgs, lib, ... }:
 
+with builtins;
+
 let
-  configfile = builtins.storePath (builtins.toFile "config" (lib.concatStringsSep "\n"
-    (map (builtins.getAttr "configLine") config.system.requiredKernelConfig))
+  removeAttrs = e: k: filter (x: x."${k}" != e."${k}");
+
+  uniqueAttrs = list: key:
+    if list == [] then []
+    else
+      let
+        x = head list;
+        xs = uniqueAttrs (lib.drop 1 list) key;
+      in [x] ++ removeAttrs x key xs;
+
+  uniqueReqKernConfig = uniqueAttrs config.system.requiredKernelConfig "configLine";
```

Comment:
Unfortunately, the `unique` must be used, or the test will fail with an error.
I changed the code, please review.